Assignment4/EderTarifa-d21c011/task07.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+<<<<<<< HEAD
+    https://colab.research.google.com/drive/1TtS8cum32KQRb6neG3aLVBpt9J4_LFrw
+=======
+    https://colab.research.google.com/github/FacultadInformatica-LinkedData/Curso2022-2023-DataScience/blob/master/Assignment4/course_materials/notebooks/Task07.ipynb
+#>>>>>>> adc606407a12d11f354d5b7f5ada676dd4a9205e
+
+**Task 07: Querying RDF(s)**
+"""
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+# TO DO
+from rdflib.plugins.sparql import prepareQuery
+
+schema = Namespace("http://www.w3.org/2000/01/rdf-schema#")
+
+q1 = prepareQuery('''
+  SELECT ?Subclass WHERE {
+    ?Subclass schema:subClassOf <http://somewhere#Person>
+  }
+  ''',
+  initNs = { "schema": schema }
+)
+# Visualize the results
+for r in g.query(q1):
+  print(r.Subclass)
+# Visualize the results
+
+from rdflib.plugins.sparql import prepareQuery
+ns = Namespace("http://somewhere#")
+
+for s1,p1,o1 in g.triples((None, RDFS.subClassOf, ns.Person)):
+  print(s1)
+
+
+q1 = prepareQuery('''
+  SELECT 
+    ?x 
+  WHERE { 
+    ?x rdfs:subClassOf ns:Person. 
+  }
+  ''',
+  initNs = { "rdfs": RDFS, "ns":ns}
+)
+
+for  t1 in g.query(q1):
+  print(t1)
+#for r in g.query(q1):
+#  print(r)
+
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+# TO DO
+syntax = Namespace('http://www.w3.org/1999/02/22-rdf-syntax-ns#')
+vcar = Namespace('http://somewhere#http://www.w3.org/2001/vcard-rdf/3.0/')
+
+q2= prepareQuery('''
+SELECT ?Fullname ?Fullname2 WHERE {
+  ?Peopleiri syntax:type <http://somewhere#Person>.
+  ?Peopleiri vcar:FN ?Fullname.
+  ?Subclassiri syntax:type <http://somewhere#Researcher>.
+  ?Subclassiri vcar:FN ?Fullname2
```

Comment:
This is not correct. You should be printing all subclasses through a query, not listing manually the subclasses

Reply:
I don't understand why do you have this query here, and then the correct one below :o